src/vs/platform/hover/browser/hover.ts

```diff
@@ -280,6 +280,9 @@ export class WorkbenchHoverDelegate extends Disposable implements IHoverDelegate
 			}));
 		}
 
+		const instantHover = this.isInstantlyHovering();
+		const id = instantHover ? (options.content instanceof HTMLElement ? options.content.innerHTML : options.content.toString()) : undefined;
```

Comment:
We should not use `innerHTML` anywhere

Reply:
You are totally right. I wasn't fully satisfied with the idea, was still thinking about this. Was considering hashing the HTML case or not having it at all. Shouldn't be to important for the time as there isn't really case of a consumer using the `WorkbenchHoverDelegate` with HTML elments so removing it.